### PR TITLE
Promise should support low level Throwable

### PR DESCRIPTION
**Pull request by WyriHaximus**

### CODE REVIEW

**Comment by asm89** [comment]
LGTM. Luckily we have `\Throwable` as typehint for `onError()`. :) https://github.com/ReactiveX/RxPHP/blob/851913a522b4955bd783d2193d7af6bda422e25d/src/ObserverInterface.php#L11

Could you add a test for this? E.g. by triggering an error by dividing by 0 or something?

**Comment by WyriHaximus** [comment]
@asm89 Updated a test

File: test/Rx/Functional/React/PromiseToObservableTest.php [hunk 1]
```diff
@@ -40,7 +40,9 @@ function () {
      */
     public function from_promise_failure()
     {
-        $p = Promise::rejected(new Exception('error'));
+        $p = new ReactPromise(function () {
+            1 / 0;
```
Comment (asm89) [inline test/Rx/Functional/React/PromiseToObservableTest.php:44]: nit: it'd be nice to add a comment here to explain we're trying to trigger an `Error`. :) I'll merge as is though.

Comment (WyriHaximus) [reply test/Rx/Functional/React/PromiseToObservableTest.php:44]: @asm89 I would have added that comment .... :D

**Pull request merged**